airflow/models/dagrun.py

```diff
@@ -466,7 +467,7 @@ def generate_run_id(run_type: DagRunType, execution_date: datetime) -> str:
     def fetch_task_instances(
         dag_id: str | None = None,
         run_id: str | None = None,
-        dag: DAG | None = None,
+        task_ids: list[str] | None = None,
```

Comment:
This is a breaking change; we cannot remove an argument from a public method. Could you deprecate it instead, or even keep it and support both params?

Reply:
Unless I am mistaken this is not breaking change. I think we can change signature of public methods as long as they are not [public interface](https://airflow.apache.org/docs/apache-airflow/stable/public-airflow-interface.html). I dont expect users to use this method in their DAGs so I think it is safe to modify it. Not being able to modify any public method would restraint very much the development in Airflow IMHO

Reply:
That's right!